integration-tests/tests/src/tests/objects.ts

```diff
@@ -17,244 +17,1323 @@
+    });
+  });
+
+  describe("boolean conversion", () => {
+    openRealmBeforeEach({ schema: [TestObjectSchema] });
+    it("converts to expected value", function (this: Mocha.Context & RealmContext) {
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean("")).equals(false, '__to_boolean("") should return false');
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean(0)).equals(false, "__to_boolean(0) should return false");
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean(-0)).equals(false, "__to_boolean(-0) should return false");
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean(null)).equals(false, "__to_boolean(null) should return false");
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean(false)).equals(false, "__to_boolean(false) should return false");
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean(NaN)).equals(false, "__to_boolean(NaN) should return false");
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean(undefined)).equals(false, "__to_boolean(undefined) should return false");
+
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean("false")).equals(true, '__to_boolean("false") should return true');
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean(1)).equals(true, "__to_boolean(1) should return true");
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean(-1)).equals(true, "__to_boolean(-1) should return true");
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean([])).equals(true, "__to_boolean([]) should return true");
+      //@ts-expect-error TYPEBUG: __to_object does not exist on realm.
+      expect(this.realm.__to_boolean(Object())).equals(true, "__to_boolean(Object()) should return true");
+    });
+  });
+
+  describe("schema", () => {
+    openRealmBeforeEach({ schema: [TestObjectSchema] });
+    it("schema fetched from object is correct type", function (this: Mocha.Context & RealmContext) {
+      const obj = this.realm.write(() => {
+        return this.realm.create<ITestObject>("TestObject", { doubleCol: 1 });
+      });
+
+      const schema = obj.objectSchema();
+      expect(schema.name).equals(TestObjectSchema.name);
+      expectArraysEqual(Object.keys(schema.properties), Object.keys(TestObjectSchema.properties));
+      expect((schema.properties.doubleCol as Realm.ObjectSchemaProperty).type).equals("double");
+    });
+  });
+  //TODO move this to notofications test
```

Comment:
Yes, please do 😄 Alternatively, if the notification tests are already covering the same methods, then we can just remove the code 🗑️

Reply:
I moved them to `notifications.ts` if you feel like they don't cohere too well with existing tests in that file, feel free to ping me and I can rewrite the object-notifications